### PR TITLE
Add broker Federator implementation

### PR DESCRIPTION
**Pull request by tpantelis**
Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

### CODE REVIEW

File: pkg/syncer/broker/federator.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		if apierrors.IsNotFound(err) {
+			removeFieldsForCreate(toDistribute)
+			toDistribute.SetNamespace(f.brokerNamespace)
+
+			klog.V(log.DEBUG).Infof("Creating resource: %#v", toDistribute)
+
+			_, err := resourceClient.Create(toDistribute, metav1.CreateOptions{})
+			if apierrors.IsAlreadyExists(err) {
+				klog.V(log.DEBUG).Infof("Resource %q already exists", toDistribute.GetName())
+				return nil
```
Comment (skitt) [inline pkg/syncer/broker/federator.go:89]: Don’t we want to update it here? (I mean, fall through to the update code.)

Comment (tpantelis) [reply pkg/syncer/broker/federator.go:89]: I thought of that but this means it was created out-of-band so unsure whether we should automatically replace it or if the caller would want to merge in this case. Perhaps we should just return the error and let the caller decide how to handle it. If they want to replace then they can retry the call. In our uses cases this `shouldn't` happen.

Comment (tpantelis) [reply pkg/syncer/broker/federator.go:89]: Thinking about this again, I think we should just update here. The caller expects its version of the resource to be applied if it succeeds and I think returning `AlreadyExists` would be confusing. This is just a timing edge case that is really equivalent to the preceding `Get` returning success a split second earlier if the resource was created out-of-band.

Comment (tpantelis) [reply pkg/syncer/broker/federator.go:89]: So I'll return a conflict error to cause a retry.

Comment (tpantelis) [reply pkg/syncer/broker/federator.go]: Done

**Pull request merged**